Android build issue - access to logs

### Godot version

4.2 master 8e0346badefcd69656e1ae3ba12d6eaafd7a4fae

### System information

Android - Windows editor

### Issue description

The root cause may be an issue with building the XR loader plugin (https://github.com/godotvr/godot_openxr_loaders) but it highlights a usability problem when diagnosing Android export issues.

Exporting gives this cryptic error:
![image](https://github.com/godotengine/godot/assets/1945449/7c3ca8bf-2fe9-4ca0-b24d-b8d545be4d34)

Checking the log in the IDE gives us:
![image](https://github.com/godotengine/godot/assets/1945449/260191a4-e63a-4ca4-b568-bb77e4cf8a83)

The error given here, "cannot remove non-existent file or directory..." is a dud, this is just a result of the export failing.
The real issue happens during the gradlew build and as the first error states "check output for the error", however Godot has closed the gradlew build window and there doesn't seem any (obvious) place to access this.

We need to either keep this window open on error and only close it on success, or we need to provide access to the last log somehow.

I ended up rerunning gradlew from the command line which gives me:
```
> Task :compileDebugJavaWithJavac FAILED
C:\Projects\godot-assets\godot_openxr_loaders\demo\android\build\src\com\godot\game\GodotApp.java:33: error: package org.godotengine.godot does not exist
import org.godotengine.godot.GodotActivity;
                            ^
C:\Projects\godot-assets\godot_openxr_loaders\demo\android\build\src\com\godot\game\GodotApp.java:41: error: cannot find symbol
public class GodotApp extends GodotActivity {
                              ^
  symbol: class GodotActivity
C:\Projects\godot-assets\godot_openxr_loaders\demo\android\build\src\com\godot\game\GodotApp.java:42: error: method does not override or implement a method from a supertype
        @Override
        ^
C:\Projects\godot-assets\godot_openxr_loaders\demo\android\build\src\com\godot\game\GodotApp.java:44: error: cannot find symbol
                setTheme(R.style.GodotAppMainTheme);
                ^
  symbol:   method setTheme(int)
  location: class GodotApp
C:\Projects\godot-assets\godot_openxr_loaders\demo\android\build\src\com\godot\game\GodotApp.java:45: error: cannot find symbol
                super.onCreate(savedInstanceState);
                ^
  symbol:   variable super
  location: class GodotApp
5 errors
```

These errors suggest an issue with the loader plugin configuration though I don't rule out an issue in the export on Godot itself.

### Steps to reproduce

- clone https://github.com/godotvr/godot_openxr_loaders
- build godot_openxr_loaders by executing `./gradlew.bat build` on the command line
- open up the demo project in this repo using Godot master
- install the android template
- run `Export Project` for the Android preset in this project (you don't need a headset to test the export)

### Minimal reproduction project

 https://github.com/godotvr/godot_openxr_loaders

--- a/editor/editor_node.cpp
+++ b/editor/editor_node.cpp
@@ -6733,7 +6733,7 @@ static void _execute_thread(void *p_ud) {
 	eta->done.set();
 }
 
-int EditorNode::execute_and_show_output(const String &p_title, const String &p_path, const List<String> &p_arguments, bool p_close_on_ok, bool p_close_on_errors) {
+int EditorNode::execute_and_show_output(const String &p_title, const String &p_path, const List<String> &p_arguments, bool p_close_on_ok, bool p_close_on_errors, String *r_output) {
 	if (execute_output_dialog) {
 		execute_output_dialog->set_title(p_title);
 		execute_output_dialog->get_ok_button()->set_disabled(true);
@@ -6779,6 +6779,9 @@ int EditorNode::execute_and_show_output(const String &p_title, const String &p_p
 		execute_output_dialog->get_ok_button()->set_disabled(false);
 	}
 
+	if (r_output) {
+		*r_output = eta.output;
+	}
 	return eta.exitcode;
 }
 

--- a/editor/editor_node.h
+++ b/editor/editor_node.h
@@ -929,7 +929,7 @@ public:
 
 	bool has_scenes_in_session();
 
-	int execute_and_show_output(const String &p_title, const String &p_path, const List<String> &p_arguments, bool p_close_on_ok = true, bool p_close_on_errors = false);
+	int execute_and_show_output(const String &p_title, const String &p_path, const List<String> &p_arguments, bool p_close_on_ok = true, bool p_close_on_errors = false, String *r_output = nullptr);
 
 	EditorNode();
 	~EditorNode();

--- a/platform/android/export/export_plugin.cpp
+++ b/platform/android/export/export_plugin.cpp
@@ -3041,10 +3041,13 @@ Error EditorExportPlatformAndroid::export_project_helper(const Ref<EditorExportP
 			}
 		}
 
-		int result = EditorNode::get_singleton()->execute_and_show_output(TTR("Building Android Project (gradle)"), build_command, cmdline);
+		String build_project_output;
+		int result = EditorNode::get_singleton()->execute_and_show_output(TTR("Building Android Project (gradle)"), build_command, cmdline, true, false, &build_project_output);
 		if (result != 0) {
-			add_message(EXPORT_MESSAGE_ERROR, TTR("Export"), TTR("Building of Android project failed, check output for the error. Alternatively visit docs.godotengine.org for Android build documentation."));
+			add_message(EXPORT_MESSAGE_ERROR, TTR("Export"), TTR("Building of Android project failed, check output for the error:") + "\n\n" + build_project_output);
 			return ERR_CANT_CREATE;
+		} else {
+			print_verbose(build_project_output);
 		}
 
 		List<String> copy_args;
@@ -3071,10 +3074,13 @@ Error EditorExportPlatformAndroid::export_project_helper(const Ref<EditorExportP
 		copy_args.push_back("-Pexport_filename=" + export_filename);
 
 		print_verbose("Copying Android binary using gradle command: " + String("\n") + build_command + " " + join_list(copy_args, String(" ")));
-		int copy_result = EditorNode::get_singleton()->execute_and_show_output(TTR("Moving output"), build_command, copy_args);
+		String copy_binary_output;
+		int copy_result = EditorNode::get_singleton()->execute_and_show_output(TTR("Moving output"), build_command, copy_args, true, false, &copy_binary_output);
 		if (copy_result != 0) {
-			add_message(EXPORT_MESSAGE_ERROR, TTR("Export"), TTR("Unable to copy and rename export file, check gradle project directory for outputs."));
+			add_message(EXPORT_MESSAGE_ERROR, TTR("Export"), TTR("Unable to copy and rename export file:") + "\n\n" + copy_binary_output);
 			return ERR_CANT_CREATE;
+		} else {
+			print_verbose(copy_binary_output);
 		}
 
 		print_verbose("Successfully completed Android gradle build.");

